pyresample/_formatting_html.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2010-2020 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Functions for html representation of area definition."""
+
+import uuid
+from functools import lru_cache
+from html import escape
+from importlib.resources import read_binary
+
+import numpy as np
+
+import pyresample.geometry as geom
+
+try:
+    import cartopy
+    cart = True
+except ModuleNotFoundError:
+    cart = False
+
+try:
+    import xarray as xr
+    from xarray.core.formatting_html import _obj_repr, datavar_section
+    xarray = True
+except ModuleNotFoundError:
+    xarray = False
+
+
+STATIC_FILES = (
+    ("pyresample.static.html", "icons_svg_inline.html"),
+    ("pyresample.static.css", "style.css"),
+)
+
+
+@lru_cache(None)
+def _load_static_files():
+    """Lazily load the resource files into memory the first time they are needed."""
+    return [
+        read_binary(package, resource).decode("utf-8")
+        for package, resource in STATIC_FILES
+    ]
+
+
+def _icon(icon_name):
+    # icon_name should be defined in pyresample/static/html/icon-svg-inline.html
+    return (
+        "<svg class='icon pyresample-{0}'>"
+        "<use xlink:href='#{0}'>"
+        "</use>"
+        "</svg>".format(icon_name)
+    )
+
+
+def plot_area_def(area, feature_res="110m", fmt="svg"):
```

Comment:
Shouldn't this, as a utility function, default to showing the matplotlib figure? Eh I guess it is in the `_formatting_html.py` module so I suppose it is fine.

Reply:
That's still a valid point. I will mark that for the refactoring also.

Reply:
Done